src/mlpack/core/data/confusion_matrix.hpp

```diff
@@ -0,0 +1,52 @@
+/**
+ * @file confusion_matrix_impl.hpp
+ * @author Jeffin Sam
+ *
+ * Compute confusion matrix to evaluate the accuracy of a classification.
+ * The function works only for discrete data/categorical data.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_CORE_DATA_CONFUSION_MATRIX_HPP
+#define MLPACK_CORE_DATA_CONFUSION_MATRIX_HPP
+
+#include <mlpack/prereqs.hpp>
+
+namespace mlpack {
+namespace data {
+
+/**
+ * A confusion matrix is a summary of prediction results on a classification
+ * problem.The number of correct and incorrect predictions are summarized
+ * with count values and broken down by each class.
+ * for example for 2 classes the function will be
+ * confusionmatrix(predictors, responses, output, 2)
+ * output matrix will be of size 2 * 2
```

Comment:
It seems like a lot of these lines aren't 80 characters, but that is easy to fix during merge.  There are also a few little grammatical errors in here (some sentences that don't start with capitalized letters, etc.).  Even though that's nitpicky, it is important for code to be consistent. :)